kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -605,14 +610,17 @@ TableInfo constructTableInfo(TableId table, com.google.cloud.bigquery.Schema big
     return tableInfoBuilder.build();
   }
 
-  private com.google.cloud.bigquery.Schema getBigQuerySchema(Schema kafkaKeySchema, Schema kafkaValueSchema) {
+  private com.google.cloud.bigquery.Schema getBigQuerySchema(Schema kafkaKeySchema, Schema kafkaValueSchema,
+                                                             com.google.cloud.bigquery.Schema bqExistingSchema) {
     com.google.cloud.bigquery.Schema valueSchema = schemaConverter.convertSchema(kafkaValueSchema);
 
     List<Field> schemaFields = intermediateTables
         ? getIntermediateSchemaFields(valueSchema, kafkaKeySchema)
         : getRegularSchemaFields(valueSchema, kafkaKeySchema);
 
-    return com.google.cloud.bigquery.Schema.of(schemaFields);
+    List<Field> validatedSchemaFields = validateAndRestoreFieldTypes(schemaFields, bqExistingSchema.getFields());
```

Comment:
Can we have a null check here and make the call to `validateAndRestoreFieldTypes` only if getFields isn't empty ? 
`if( bqExistingSchema != null && !bqExistingSchema.getFrields().isEmpty())`